application/src/main/kotlin/net/corda/v5/application/membership/NotaryLookup.kt

```diff
@@ -0,0 +1,69 @@
+package net.corda.v5.application.membership
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.types.MemberX500Name
+import net.corda.v5.membership.NotaryInfo
+import java.security.PublicKey
+
+/**
+ * [NotaryLookup] allows flows to retrieve the [NotaryInfo] in the network.
+ *
+ * The platform will provide an instance of [NotaryLookup] to flows via property injection.
+ */
+@DoNotImplement
+interface NotaryLookup {
+
+    /**
+     * A list of notary services available on the network.
+     *
+     * Note that the services are sorted based on legal name, and the
+     * ordering might change once new notaries are introduced.
+     *
+     * In case of converting a returned list to a string representation
+     * use another delimiter rather than "," or wrap each entry with a
+     * postfix and a prefix:
+     * ```
+     * list.joinToString("|")
+     * list.map { "[$it]" }.toString()
+     * ```
+     * This allows to distinguish between each element of a map, as
+     * [MemberX500Name.toString] returns comma separated values of [MemberX500Name].
+     *
+     * @return A list of the network's Notary services.
+     */
+    @Suspendable
```

Comment:
I'm curious why is this required?

Reply:
It seem like this is required on all functions based on the conversations in the runtime os PR

Reply:
Added